### ISSUE TITLE
[CI] Integrate container image scanning into Meshery's CI pipeline

### ISSUE THREAD

**Issue by s1ntaxe770r** [meshery/meshery#4573]
#### Current Behavior

Currently ,  Meshery's release pipeline does not include security scanning for Docker images. This is not exactly ideal, what's the point of writing secure code when the underlying host is vulnerable. Asides from that manually patching or updating container images is a chore
#### Desired Behavior

I propose we take a look at existing tools and discuss the best approach to integrating this security scanning into our release pipelines.
#### Implementation

We can take a look at  [Trivy](https://github.com/aquasecurity/trivy) or [Hadolint](https://github.com/hadolint/hadolint) for lining and security scanning. I'm open to suggestions about what tools to use or if this is worth pursuing right now. 
#### Acceptance Tests

To be determined..
---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

**Comment by leecalcote** [comment]
Good suggestion. This isn't the first time this topic has come up. There may well be an existing issue on this, actually. We should de-duplicate, if so.

Also, note the scanning going on here - https://artifacthub.io/packages/helm/meshery/meshery?modal=security-report

**Comment by leecalcote** [comment]
After a quick search, I don't see a duplicate issue. ✔️

**Comment by leecalcote** [comment]
We’re also getting a number of security reports from CodeQL. Let's see about getting you access to those.

**Comment by RahulSurwade08** [comment]
Hey @leecalcote I would love to contribute for this issue. How can i get started?

**Comment by leecalcote** [comment]
@RahulSurwade08, great. I'll call on @Aisuko to assist.

**Comment by RahulSurwade08** [comment]
Sure Thanks. @leecalcote

**Comment by RahulSurwade08** [comment]
@leecalcote Any update here?

**Comment by acald-creator** [comment]
@leecalcote 

I didn't see this issue already open for container image scanning.

The only question I had about this was that since you're already using ArtifactHub for the vulnerabilities reports, but I'm not sure if there is already a central location within this project for scan reports, either from trivy or something similar.

I can do research on this, if no one else hasn't done anything yet.

**Comment by huntingcodes-001** [comment]
Good catch! To address this, I propose integrating Trivy for vulnerability scanning and Hadolint for Dockerfile linting within the CI/CD pipeline. 

@leecalcote kindly assign the taks to me
I'll take this up and solve the issue!

**Comment by chilkaditya** [comment]
Hi @leecalcote I came across this issue through the KubeCon EU 2025 Security Slam initiative and was wondering if it's open for contributions. I'm interested in contributing to security-focused projects and would like to help move this one toward achieving the Security Sentinel badge.
